### PR TITLE
Construct the skreleton

### PR DESCRIPTION
**Pull request by trentwirth**
created a dictionary of mediapipe++ joints from qualisys data (`.tsv`) input

the markers & weights for certain joints will need to be updated (joint composition and weight is recorded in `dumb_qualisys_joint_center_dictionary_builder.py`) 

I created a plotting function (`skelly_plotter.py`) to look at the joint data afterward, but this could definitely use some TLC from Jon and maybe Aaron.

### CODE REVIEW

**Comment by jonmatthis** [comment]
Need to update the ReadMe with: 

 - Installation instructions
 - Run instructions
 - URL to demo data to download

File: proccess_data.py [hunk 1]
```diff
@@ -13,4 +14,10 @@ def process_data(subject_qualisys_json_path: Path,
 
     qualisys_dict = qualisys_json_creator(subject_qualisys_json_path, qualisys_df)
 
-    skeleton_frame_marker_dimension_npy = qualisys_to_generic_skeleton_converter(qualisys_dict)
+    generic_skelly_dict = qualisys_to_generic_skeleton_converter(qualisys_dict)
+
+    select_frame = np.array([1500])
```
Comment (jonmatthis) [inline proccess_data.py]: what's this number? Can it be specified in `main.py`?

File: requirements.txt [hunk 2]
```diff
@@ -1,15 +1,11 @@
 numpy==1.23.2
```
Comment (jonmatthis) [inline requirements.txt:1]: clean up all these imports

Only add imports that YOU PERSONALLY are importing in your files

File: utilities/dumb_qualisys_joint_center_dictionary_builder.py [hunk 3]
```diff
@@ -0,0 +1,37 @@
...
+    # into a generic meadiapipe++ format
+
+    # below is a handwritten dictionary, where the key represents the joint center name
+    # that is going into the skeleton, the first value represents the physical qualisys
+    # markers that are being combined to be put into the joint center, and the second value
+    # (or array of values) is the relative weight of each of the markers for the joint
+
+    qualisys_joint_center_dict = {
+        "head_xyz": [["HeadL+HeadR"], [0.5, 0.5]],
+        "left_head_marker_xyz": [["HeadL"], [1]],
```
Comment (jonmatthis) [inline utilities/dumb_qualisys_joint_center_dictionary_builder.py]: Formatting is kinda wrong - You're making a 'dictionary full of kinda janky lists' when you should be making a 'dictionary of dictionaries'

File: utilities/dumb_qualisys_joint_center_dictionary_builder.py [hunk 4]
```diff
@@ -0,0 +1,37 @@
...
+
+    # this is a special purpose dictionary for getting ARGP skeleton data from qualisys
+    # into a generic meadiapipe++ format
+
+    # below is a handwritten dictionary, where the key represents the joint center name
+    # that is going into the skeleton, the first value represents the physical qualisys
+    # markers that are being combined to be put into the joint center, and the second value
+    # (or array of values) is the relative weight of each of the markers for the joint
+
+    qualisys_joint_center_dict = {
```
Comment (jonmatthis) [inline utilities/dumb_qualisys_joint_center_dictionary_builder.py]: We should consider switch to making these definitinos in a `yaml` file, which also loads nicely as a `dictionary` but with much friendlier, `markdown`-friendly synctax: 

see: https://www.cloudbees.com/blog/yaml-tutorial-everything-you-need-get-started

File: utilities/get_marker_key_list_for_joint.py [hunk 5]
```diff
@@ -0,0 +1,13 @@
+import string
+
+
+def get_marker_key_list_for_joint(qualisys_dict: dict,
```
Comment (jonmatthis) [inline utilities/get_marker_key_list_for_joint.py:4]: pretty sure this function isn't necessary - dictionaries can return their keys directly as an interable

File: utilities/get_joint_from_markers.py [hunk 6]
```diff
@@ -0,0 +1,44 @@
...
+    marker_dict = {}  # create empty dictionary to update
+
+    for single_marker_name in marker_string_names:
+        # get the keys for a joint
+        marker_key_list = get_marker_key_list_for_joint(qualisys_dict, single_marker_name)
+
+        marker_array = np.array([])
+
+        # use the keys to assemble an np.array to put inside a dictionary
+        for key in marker_key_list:
```
Comment (jonmatthis) [inline utilities/get_joint_from_markers.py]:
```python
for key in dictionary_thing.keys():
   print(f"this key is: {key}")
   
   ```

File: utilities/get_marker_key_list_for_joint.py (hunk 5 above)
Comment (jonmatthis) [reply utilities/get_marker_key_list_for_joint.py:4]: see code sample above

File: main.py [hunk 7]
```diff
@@ -36,3 +36,5 @@
     process_data(subject_qualisys_json_path, qualisys_df, pupil_df)
```
Comment (jonmatthis) [inline main.py]: separate "process_data" from "plot_data" functionalities

that is - 

First - Do all the processing and save the data out in whatever kinda way (i.e. in `process_data` function), 

THEN - Have your plotting method load in that data and try to make a visualization out of it

**Review by jonmatthis — CHANGES_REQUESTED** [review]

File: utilities/dumb_qualisys_joint_center_dictionary_builder.py (hunk 4 above)
Comment (jonmatthis) [reply utilities/dumb_qualisys_joint_center_dictionary_builder.py]: ...or maybe not - There's apparently a whole debate on the 'YAML' vs 'TOML' thing, and it kinda makes me want to stick to 'JSON' for now 😂

File: proccess_data.py (hunk 1 above)
Comment (trentwirth) [reply proccess_data.py]: this number is the "favorite frame" for debugging, it's been moved to `main.py`

File: requirements.txt (hunk 2 above)
Comment (trentwirth) [reply requirements.txt:1]: imports are cleaned

File: utilities/dumb_qualisys_joint_center_dictionary_builder.py (hunk 3 above)
Comment (trentwirth) [reply utilities/dumb_qualisys_joint_center_dictionary_builder.py]: formatting is fixed!

File: main.py (hunk 7 above)
Comment (trentwirth) [reply main.py]: separated ✔️

**Comment by trentwirth** [comment]
oh, I just re-requested review but I forgot to add to the readme, I'll do that and then re-request

**Comment by trentwirth** [comment]
I've updated the `ReadMe.md` complete with installation & run instructions, plus demo data linked through my NEU google drive account

**Pull request merged**